rest/config_test.go

```diff
@@ -1219,3 +1221,200 @@ func TestExpandEnv(t *testing.T) {
+		configFile := createTempFile(t, []byte(`{"unknownKey":"unknownValue"}`))
+		defer deleteTempFile(t, configFile)
+		args := []string{"sync_gateway", configFile.Name()}
+		config, err := setupServerConfig(args)
+		require.Error(t, err, "Should throw error reading file")
+		assert.Contains(t, err.Error(), "unrecognized JSON field")
+		assert.Nil(t, config)
+	})
+
+	t.Run("Run setupAndValidate with a config file that doesn't exist", func(t *testing.T) {
+		configFile := createTempFile(t, []byte(``))
+		args := []string{"sync_gateway", configFile.Name()}
+		deleteTempFile(t, configFile)
+		config, err := setupServerConfig(args)
+		require.Error(t, err, "Should throw error reading file")
+		assert.Contains(t, err.Error(), "Error reading config file")
+		assert.Nil(t, config)
+	})
+
+	t.Run("Run setupAndValidate with illegal value for stats_log_freq_secs", func(t *testing.T) {
+		configFile := createTempFile(t, []byte(`
+		{
+		  "databases": {
+		    "db": {
+		      "bucket": "leaky_bucket",
+		      "server": "couchbase://localhost",
+		      "username": "Administrator",
+		      "password": "password"
+		    }
+		  },
+		  "unsupported": {
+		    "stats_log_freq_secs": 1
+		  }
+		}`))
+		defer deleteTempFile(t, configFile)
+		args := []string{"sync_gateway", configFile.Name()}
+		config, err := setupServerConfig(args)
+		require.Error(t, err, "Should throw error reading file")
+		assert.Contains(t, err.Error(), "minimum value for unsupported.stats_log_freq_secs is: 10")
+		assert.Nil(t, config)
+	})
+
+}
+
+func TestSetupServerContext(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyAll)()
+	t.Run("Create server context with a valid configuration", func(t *testing.T) {
+		config := &ServerConfig{
+			Logging: &base.LoggingConfig{
+				RedactionLevel: base.RedactNone,
+			},
+		}
+		sc, err := setupServerContext(config)
+		require.NoError(t, err)
+		require.NotNil(t, sc)
+		sc.Close()
+	})
+
+	t.Run("Create server context with BcryptCost inside allowed range", func(t *testing.T) {
+		config := &ServerConfig{
+			Logging: &base.LoggingConfig{
+				RedactionLevel: base.RedactFull,
+			},
+			BcryptCost: bcrypt.MaxCost,
+		}
+		defer func() { require.NoError(t, auth.SetBcryptCost(bcrypt.DefaultCost)) }()
+		sc, err := setupServerContext(config)
+		require.NoError(t, err)
+		require.NotNil(t, sc)
+		sc.Close()
+	})
+
+	t.Run("Create server context with BcryptCost outside allowed range", func(t *testing.T) {
+		config := &ServerConfig{
+			Logging: &base.LoggingConfig{
+				RedactionLevel: base.RedactPartial,
+			},
+			BcryptCost: bcrypt.MaxCost + 1,
+		}
+		defer func() { require.NoError(t, auth.SetBcryptCost(bcrypt.DefaultCost)) }()
+		sc, err := setupServerContext(config)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "configuration error: 32 outside allowed range: 10-31: invalid bcrypt cost")
+		assert.Nil(t, sc)
+	})
+}
+
+func TestStartServer(t *testing.T) {
```

Comment:
As discussed, can remove, startServer gets sufficient coverage when running SG outside of unit test framework.

Reply:
Sure Adam, this has been removed in the latest revision.